scenes/quests/lore_quests/quest_001/1_music_puzzle/components/trees.gd

```diff
@@ -0,0 +1,15 @@
+# SPDX-FileCopyrightText: The Threadbare Authors
+# SPDX-License-Identifier: MPL-2.0
+@tool
+extends Node2D
+
+const WOOL_PERCENTAGE: float = 0.4
+
+const TREE_WOOL_GREEN_01 = preload(
+	"res://scenes/game_elements/props/tree/components/Tree_Wool_Green_01.png"
+)
+const TREE = preload("res://scenes/game_elements/props/tree/components/Tree.png")
+
+@export_tool_button("Randomize Trees Appearances") var a = func():
+	for child in get_children():
+		child.set("tree__texture", TREE if randf() > WOOL_PERCENTAGE else TREE_WOOL_GREEN_01)
```

Comment:
This works for now. If we add more weighted choices it would be better to use [RandomNumberGenerator.rand_weighted()](https://docs.godotengine.org/en/stable/tutorials/math/random_number_generation.html#weighted-random-probability). Example:

```
var fruits = ["apple", "orange", "pear", "banana"]
var probabilities = [0.5, 1, 1, 2];

var random = RandomNumberGenerator.new()
print(fruits[random.rand_weighted(probabilities)])
```